prow/pod-utils/decorate/podspec.go

```diff
@@ -705,10 +705,24 @@ func decorate(spec *coreapi.PodSpec, pj *prowapi.ProwJob, rawEnv map[string]stri
 		spec.Containers[i].Env = append(container.Env, KubeEnv(rawEnv)...)
 	}
 
+	secretVolumes := sets.NewString()
+	for _, volume := range spec.Volumes {
+		if volume.VolumeSource.Secret != nil {
+			secretVolumes.Insert(volume.Name)
+		}
+	}
+	containsSecretData := func(volumeName string) bool {
+		if censor := pj.Spec.DecorationConfig.CensorSecrets; censor != nil && !*censor {
```

Comment:
Nit: A tad confusing to read, would be easier as:
```suggestion
		if censor := pj.Spec.DecorationConfig.CensorSecrets; censor == nil || !*censor {
```